common/utils.js

```diff
@@ -3,7 +3,22 @@
 
     angular.module('chaise.utils', [])
 
-    .factory('UriUtils', ['$injector', '$window', 'parsedFilter', '$rootScope', function($injector, $window, ParsedFilter, $rootScope) {
+    .constant("appTagMapping", {
+        "tag:isrd.isi.edu,2016:chaise:record": "record",
+        "tag:isrd.isi.edu,2016:chaise:record-two": "record-two",
+        "tag:isrd.isi.edu,2016:chaise:viewer": "viewer",
+        "tag:isrd.isi.edu,2016:chaise:search": "search",
+        "tag:isrd.isi.edu,2016:chaise:recordset": "recordset",
+        "tag:isrd.isi.edu,2016:chaise:recordedit": "recordedit"
+})
+
+    .factory('UriUtils', ['$injector', '$window', 'parsedFilter', '$rootScope', 'appTagMapping', function($injector, $window, ParsedFilter, $rootScope, appTagMapping) {
+
+        var chaiseBaseURL = $window.location.href.replace($window.location.hash, '');
+        var apps = ['record', 'recordset', 'record-two', 'search', 'viewer'];
+        apps.forEach(function(app) {
```

Comment:
I'm not sure I understand the point of this loop. 

If you are looping through the array to replace the url based on the app the user is on, we should be attaching that information to `$rootScope.context.app`. And you could just replace based on that value instead of looping through this array.